extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfig.java

```diff
@@ -107,6 +107,12 @@
     @ConfigItem(defaultValue = "ClusterIP")
     ServiceType serviceType;
 
+    /**
+     * The nodePort to set when serviceType is set to nodePort
+     */
+    @ConfigItem(defaultValue = "31987")
```

Comment:
Is there a way to _not_ specify the concrete nodePort and let Kubernetes assign a port automatically? Perhaps allow the 0 value, which typically means "port should be auto-assigned"?

Reply:
When you don't set it, it gets assigned automatically. The problem is that it gets **re**-assigned to another value when you update the application and that's what I am trying to avoid.

Reply:
Sure. What I'm trying to say is: is there a way how to get back to the default behavior?

Reply:
Yeah, that essentially boils down to what @iocanel  is asking for :)